File: Dataset/cs/Filecopy.cs

//  Filecopy.cs  (c) 2003 Kari Laitinen

//  www.naturalprogramming.com

using System ;
using System.IO ;  // Classes for file handling.

class Filecopy
{
   static void Main()
   {
      Console.Write( "\n This program copies text from one file"
                   + "\n to another file. Please, give the name of"
                   + "\n the file to be copied: " ) ;

      string  name_of_file_to_be_copied  =  Console.ReadLine() ;

      Console.Write( "\n Give the name of the duplicate file: " ) ;

      string  name_of_new_duplicate_file  =  Console.ReadLine() ;

      try
      {
         StreamReader file_to_be_copied  =
                          new  StreamReader( name_of_file_to_be_copied ) ;

         StreamWriter new_duplicate_file  =
                          new  StreamWriter( name_of_new_duplicate_file ) ;

         Console.Write( "\n Copying in progress ... \n" ) ;

         int  text_line_counter  =  0 ;
         bool end_of_file_encountered  =  false ;

         while ( end_of_file_encountered  ==  false )
         {
            string  text_line_from_file  =  file_to_be_copied.ReadLine() ;

            if ( text_line_from_file  ==  null )
            {
               end_of_file_encountered  =  true ;
            }
            else
            {
               new_duplicate_file.WriteLine( text_line_from_file ) ;
               text_line_counter  ++  ;
            }
         }

         Console.Write( "\n Copying ready. "  +  text_line_counter
                      + " lines were copied. \n" ) ;

         file_to_be_copied.Close() ;
         new_duplicate_file.Close() ;
      }
      catch ( FileNotFoundException )
      {
         Console.Write( "\n Cannot open "  +  name_of_file_to_be_copied ) ;
      }
      catch ( Exception )
      {
         Console.Write( "\n File error. Probably cannot write to "
                     +  name_of_new_duplicate_file ) ;
      }
   }
}

File: Dataset/cpp/filecopy.cpp

//  filecopy.cpp  (c) 1997-2002 Kari Laitinen

#include  <iostream.h>
#include  <fstream.h>  // File handling classes.

int main()
{
   ifstream  file_to_be_copied ;
   ofstream  new_duplicate_file ;

   char  name_of_file_to_be_copied[ 50 ] ;
   char  name_of_new_duplicate_file[ 50 ] ;

   cout << "\n This program copies text from one file"
        << "\n to another file. Please, give the name of"
        << "\n the file to be copied: " ;

   cin.getline( name_of_file_to_be_copied,
                sizeof( name_of_file_to_be_copied ) ) ;

   cout << "\n Give the name of the duplicate file: " ;

   cin.getline( name_of_new_duplicate_file,
                sizeof( name_of_new_duplicate_file ) ) ;

   file_to_be_copied.open( name_of_file_to_be_copied ) ;
   new_duplicate_file.open( name_of_new_duplicate_file ) ;

   if ( file_to_be_copied.fail() )
   {
      cout << "\n Cannot open "  <<  name_of_file_to_be_copied ;
   }
   else if ( new_duplicate_file.fail() )
   {
      cout << "\n Cannot open "  <<  name_of_new_duplicate_file ;
   }
   else
   {
      char text_line_from_file[ 150 ] ;
      int  text_line_counter  =  0 ;

      cout << "\n Copying in progress ... \n" ;

      while( ! file_to_be_copied.eof() )
      {
         file_to_be_copied.getline( text_line_from_file,
                                    sizeof( text_line_from_file) );

         new_duplicate_file  <<  text_line_from_file ;

         if ( ! file_to_be_copied.eof() )
         {
            new_duplicate_file  <<  "\n" ;
         }

         text_line_counter  ++  ;
      }

      cout << "\n Copying ready. "  <<  text_line_counter
           << " lines were copied. \n" ;
   }
}



